### PR TITLE
Fix Stuck Beacon Node

### LINKED ISSUE #4482: Beacon Node Stuck in Sync
Our beacon nodes frequently get stuck when syncing with the testnet from blocks `217 - 256`. The node endlessly requests that range of blocks and never seems to move past it, while CPU usage is close to 100%. @nisdas hypothesized there is some issue with bolt being resource constrained while in a docker environment due to how it does MMAP system calls and how it allocates data. Changing `boltDB.AllocSize = 8 * 1024 * 1024` and clearing a node DB fixed the issue for me locally and was able to sync. We believe this is related to the root cause of what triggered #4476.

### PR DESCRIPTION
**Pull request by nisdas**
- [x] Reverts prysmaticlabs/prysm#4449
- [x] Request blocks from the 1st deposit log instead of the 1st eth1 block.
- [x] Add contract deployment block flag.

Resolves #4482

### CODE REVIEW

File: shared/params/config.go [hunk 1]
```diff
@@ -96,6 +96,7 @@ type BeaconChainConfig struct {
 	DefaultPageSize           int           // DefaultPageSize defines the default page size for RPC server request.
 	MaxPageSize               int           // MaxPageSize defines the max page size for RPC server respond.
 	MaxPeersToSync            int           // MaxPeersToSync describes the limit for number of peers in round robin sync.
+	ContractDeploymentBlock   int           // ContractDeploymentBlock defines eth1 block in which the dpeosit contract was deployed/
```
Comment (terencechain) [inline shared/params/config.go]:
```suggestion
	ContractDeploymentBlock   int           // ContractDeploymentBlock defines eth1 block in which the dpeosit contract was deployed.
```

File: beacon-chain/powchain/testing/mock.go [hunk 2]
```diff
@@ -6,7 +6,12 @@ import (
 	"math/big"
 	"time"
 
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+
```
Comment (terencechain) [inline beacon-chain/powchain/testing/mock.go]:
```suggestion
```

File: beacon-chain/powchain/service.go [hunk 3]
```diff
@@ -448,6 +456,45 @@ func (s *Service) processSubscribedHeaders(header *gethTypes.Header) {
 	}
 }
 
+// batchRequestHeaders requests the block range specified in the arguments. Instead of requesting
+// each block in one call, it batches all requests into a single rpc call.
+func (s *Service) batchRequestHeaders(startBlock uint64, endBlock uint64) ([]*gethTypes.Header, error) {
```
Comment (terencechain) [inline beacon-chain/powchain/service.go]: Should we add a test for this?

Comment (nisdas) [reply beacon-chain/powchain/service.go]: this is actually difficult to test for. We use the raw rpc call over here to batch request all blocks here, so I can't test it out using the simulated backend.

File: beacon-chain/powchain/service.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	gethTypes "github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
```
Comment (prestonvanloon) [inline beacon-chain/powchain/service.go]: Duplicate import?

File: beacon-chain/powchain/log_processing_test.go [hunk 5]
```diff
@@ -9,6 +9,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/flags"
+
```
Comment (prestonvanloon) [inline beacon-chain/powchain/log_processing_test.go]: Gap

File: beacon-chain/powchain/service.go (hunk 3 above)
Comment (terencechain) [reply beacon-chain/powchain/service.go]: Fair 👍🏼

File: shared/params/config.go [hunk 6]
```diff
@@ -188,6 +189,7 @@ var defaultBeaconConfig = &BeaconChainConfig{
 	DefaultPageSize:           250,
 	MaxPageSize:               500,
 	MaxPeersToSync:            15,
+	ContractDeploymentBlock:   1960177,
```
Comment (prestonvanloon) [inline shared/params/config.go]: This is never used? Looks like you only use the flag value.

File: beacon-chain/powchain/block_reader_test.go [hunk 7]
```diff
@@ -42,6 +43,8 @@ func TestLatestMainchainInfo_OK(t *testing.T) {
 		t.Fatalf("unable to setup web3 ETH1.0 chain service: %v", err)
 	}
 	web3Service = setDefaultMocks(web3Service)
+	web3Service.rpcClient = &mockPOW.RPCClient{Backend: testAcc.Backend}
+
```
Comment (shayzluf) [inline beacon-chain/powchain/block_reader_test.go]:
```suggestion
```

**Pull request merged**